src/software/world/timestamped_possession_state.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <map>
+#include <optional>
+
+#include "software/time/timestamp.h"
+#include "software/world/robot_state.h"
+#include "software/world/team_types.h"
+
+/**
+ * A light structure for a robot ID with TeamSide
+ */
+struct RobotIdWithTeamSide
+{
+    unsigned int id;
+    TeamSide team_side;
+
+    bool operator==(const RobotIdWithTeamSide &other) const
+    {
+        return id == other.id && team_side == other.team_side;
+    }
+
+    bool operator<(const RobotIdWithTeamSide &other) const
+    {
+        return id < other.id || team_side < other.team_side;
+    }
+};
+
+class TimestampedPossessionState
```

Comment:
Couple questions about this class / its assumptions
* Can more than one robot (of all robots on the field) have possession at a time?
* Can each team have possession at once? (ie. one friendly and one enemy robot both have possession)?

Reply:
* more than one robot can have possession (imagine two robots fighting over the ball)
* if both teams have robots that have possession then getTeamWithPossession will return std::nullopt

Reply:
> if both teams have robots that have possession then getTeamWithPossession will return std::nullopt

nit, and this depends exactly how we want to use possession, but perhaps `getTeamWIthExclusivePossession` would be a more clear name then? Otherwise I don't think it would be obvious if the case you mentioned where two robots are fighting for a ball counts as possession for a team or not.

Reply:
added Exclusive